eth/beacon/_utils/hash.py

```diff
@@ -1,8 +1,12 @@
+from typing import (
+    Hashable,
+)
+
 from eth_typing import Hash32
 from eth_hash.auto import keccak
 
 
-def hash_eth2(data: bytes) -> Hash32:
+def hash_eth2(data: Hashable) -> Hash32:
```

Comment:
I might miss some historical discussion. But is `Hashable` what we want here?
According to typing documentation, `Hashable` is an alias to `collections.abc.Hashable`, which checks if an object has the abstract method `__hash__`. That is in the sense of Python `hash` rather then if a `Serializable` object could be hash_eth2-ed or tree-hashed.